docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+  // signed is a signature of the message
+  bytes signed  = 2 [(gogoproto.nullable) = false];
+
+  // TODO: TBD
+  // msg_seq helps to keep track over multiple handshakes, will by set by the sender
+  uint64 msg_seq = 1 [(gogoproto.nullable) = false];
+  // peer_id is the peer that the message is targeted for, will by set by the sender
+  bytes peer_id  = 2 [(gogoproto.nullable) = false];
+}
+
+// NodeInfo contains node's information
+message NodeInfo {
+  // peer_id of the authenticating node
+  bytes peer_id          = 1 [(gogoproto.nullable) = false];
+  // operator_id of the authenticating node
+  bytes operator_id      = 2 [(gogoproto.nullable) = true];
+  // node_type is the type of the authenticating node
+  uint64 node_type       = 3 [(gogoproto.nullable) = false];
+  // execution_node is the "name/version" of the eth1 node used by the node
+  string execution_node  = 4; // TBD: actual value
+  // consensus_node is the "name/version" of the eth2 node used by the node
+  string consesnsus_node = 5; // TBD: actual value
+  // fork_version is the current fork used by the node
+  uint32 fork_version    = 6;
+  // fork_version is the current ssv-node version
+  string node_version    = 7;
+}
+
+```
+
+---
+
+
+## Networking
+
+### Pubsub
+
+The main purpose is for broadcasting messages to a group (AKA subnet) of nodes. \
+In addition, the following are achieved as well:
+
+- subscriptions metadata helps to get liveliness information of nodes
+- pubsub scoring enables to prune bad/malicious peers based on network behavior and application-specific rules
+
+The following parameters are used for initializing pubsub:
+
+- `floodPublish` was turned on for better reliability, as peer's own messages will be propagated to a larger set of peers 
+  (see [Flood Publishing](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#flood-publishing)) 
+- `peerOutboundQueueSize` / `validationQueueSize` were increased to `600`, to avoid dropped messages on bad connectivity or slow processing
+- `directPeers` includes the exporter peer ID, to ensure it gets all messages
+- `subscriptionFilter` was injected to ensure a peer will connect to relevant topics, see [SubscriptionFilter](https://github.com/libp2p/go-libp2p-pubsub/blob/master/subscription_filter.go) interface
+- (fork `v1`) `msgID` is a custom function that calculates a `msg-id` based on the message content hash. 
+The default function uses the `sender` + `msg_seq` which we don't track, and enforces signature / verification for each message. 
+As all the messages are being verified using the share key, it would be redundant to it also in the pubusb level.
+Moreover, the default `msg-id` duplicates messages, causing it to be processed more than once, in case it was sent by multiple peers (e.g. decided message).
+- (fork `v1`) `signPolicy` was set to `StrictNoSign` (required for custom `msg-id`) to avoid producing and verifying message signatures in the pubsub router
```

Comment:
it means messages are not signed with the network private key?

Reply:
Pubsub is based on streams, and the network key is used to establish secured connections/streams ([go-libp2p-noise](https://github.com/libp2p/go-libp2p-noise)).

The idea is to reduce to work done in pubsub level, as pubsub router by default signs and verifies messages (https://github.com/libp2p/specs/blob/master/pubsub/README.md#message-signing)